### PR TITLE
runtime: unlock the alive lock only once

### PR DESCRIPTION
**Pull request by vrothberg**
Unlock the alive lock only once in the deferred func call.

Fixes: #3207
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@giuseppe PTAL

File: libpod/runtime.go [hunk 1]
```diff
@@ -891,7 +888,6 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (err error) {
 		// no containers running.  Create immediately a namespace, as
 		// we will need to access the storage.
 		if os.Geteuid() != 0 {
-			aliveLock.Unlock()
```
Comment (giuseppe) [inline libpod/runtime.go]: I think think will cause a deadlock.  BecomeRootInUserNS will cause a re-exec, and will wait for the new process to terminate.  But that won't happen as the new process will also try to get a lock on the "alive" file

Comment (vrothberg) [reply libpod/runtime.go]: Ouch, that sounds like deadlock. Thanks for the hint! I'll fix that up and ping after repush.

Comment (giuseppe) [reply libpod/runtime.go]: do we need to set `locked = false` to avoid the double unlock?

Comment (vrothberg) [reply libpod/runtime.go]: Should be better now. WDYT?

**Comment by mheon** [comment]
Code LGTM. Gating's exploding though.

File: libpod/runtime.go (hunk 1 above)
Comment (giuseppe) [reply libpod/runtime.go]: yes, thanks for the fix.  LGTM

**Comment by vrothberg** [comment]
Given the two LGTMs...
/lgtm

**Pull request merged**